crates/sql/src/ast/mod.rs

```diff
@@ -0,0 +1,309 @@
+        match (iter.next(), iter.next()) {
+            (Some(table), Some(column)) => SqlColumn {
+                column: column.into(),
+                table: Some(table.into()),
+                alias: None,
+            },
+            (Some(column), None) => SqlColumn {
+                column: column.into(),
+                table: None,
+                alias: None,
+            },
+            _ => panic!("{}", &format!("Invalid column: `{value}`")),
+        }
+    }
+}
+
+impl From<Ident> for SqlColumn {
+    fn from(value: Ident) -> Self {
+        SqlColumn {
+            column: value.into(),
+            table: None,
+            alias: None,
+        }
+    }
+}
+
+impl From<Vec<Ident>> for SqlColumn {
+    fn from(value: Vec<Ident>) -> Self {
+        match &value[..] {
+            [table, column] => SqlColumn {
+                column: column.clone().into(),
+                table: Some(table.clone().into()),
+                alias: None,
+            },
+            [column] => SqlColumn {
+                column: column.clone().into(),
+                table: None,
+                alias: None,
+            },
+            _ => panic!("{}", &format!("Invalid column: `{value:?}`")),
+        }
+    }
+}
+
+impl fmt::Display for SqlColumn {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let alias = match &self.alias {
+            Some(alias) => format!(" AS {}", alias),
+            None => "".to_string(),
+        };
+        match &self.table {
+            Some(table) => write!(f, "{}.{}{alias}", table, self.column),
+            None => write!(f, "{}{alias}", self.column),
+        }
+    }
+}
+
+/// Helper to override the `Display` trait for `OpQuery` enum,
+/// to convert them to their `SQL` representation.
+#[derive(Debug)]
+pub struct PrintOperator {
+    pub(crate) op: OpQuery,
+}
+
+impl fmt::Display for PrintOperator {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.op {
+            OpQuery::Cmp(op) => match op {
+                OpCmp::Eq => {
+                    write!(f, "=")
+                }
+                OpCmp::NotEq => {
+                    write!(f, "<>")
+                }
+                x => {
+                    write!(f, "{}", x)
+                }
+            },
+            OpQuery::Logic(op) => match op {
+                OpLogic::And => {
+                    write!(f, "AND")
+                }
+                OpLogic::Or => {
+                    write!(f, "OR")
+                }
+            },
+        }
+    }
+}
+
+/// Raw `SQL` value.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum SqlValue {
+    /// Boolean value
+    Bool(bool),
+    /// Hexadecimal value
+    /// ```sql
+    /// 0x0000FCAB
+    /// x'93dda0'
+    /// ```
+    Hex(String),
+    /// Unparsed number (int, float, etc.)
+    Num(String),
+    /// Unparsed long number (bigint, etc.)
+    NumLong(String),
```

Comment:
We should probably just have a single variant. The exact int/float type will be determined during type checking.